core/dronecore.cpp

```diff
@@ -23,6 +23,76 @@ DroneCore::~DroneCore()
     _impl = nullptr;
 }
 
+DroneCore::ConnectionResult DroneCore::add_any_connection(std::string str)
+{
+    std::string delimiter = "://";
+    std::string connection_str[3];
+    size_t pos = 0, i = 0;
+    while(i<2)
+    {
+        pos = str.find(delimiter);
+        if(pos != std::string::npos)
+        {
+            connection_str[i] = str.substr(0, pos);
+            str.erase(0, pos+delimiter.length());
+            if(str == "")
+				break;
+            delimiter=":";
+        }
+	    i++;
+    }
+    connection_str[2] = str;
+    if(connection_str[0] != "serial")
+        return add_link_connection(connection_str);
+    else if(connection_str[0] == "serial")
+    {
+        if(connection_str[1] == "")
+            return add_serial_connection();
+        else
+            return add_serial_connection(connection_str[1], stoi(connection_str[2]));
+    }
+	return DroneCore::ConnectionResult::DESTINATION_IP_UNKNOWN;
```

Comment:
Is this the correct error to return? Do we need another one?

Reply:
it's actually better to add a new error. May be, URL_INVALID